evalml/pipelines/utils.py

```diff
@@ -250,7 +250,12 @@ def _get_decomposer(X, y, problem_type, estimator_class, sampler_name=None):
             return components
         freq = time_index.freq.name
         if STLDecomposer.is_freq_valid(freq):
-            components.append(STLDecomposer)
+            # Make sure there's a seasonal period
+            order = 3 if "Q" in freq else 5
+            temp_decomp = STLDecomposer(time_index, rel_max_order=order)
+            seasonal_period = temp_decomp.determine_periodicity(X, y)
```

Comment:
if we're utilizing `determine_periodicity` like this - maybe it should be a class method?